infra/relayer/src/purse.js

```diff
@@ -0,0 +1,696 @@
+     * error message that makes no sense.
+     */
+    const signed = await this.web3.eth.accounts.signTransaction(
+      unsigned,
+      bufferToHex(masterPrivkey)
+    )
+    const rawTx = signed.rawTransaction
+
+    // The following isn't availible in our pinned version of web3.js...
+    // const txHash = signed.transactionHash
+    // So we hash it ourself...
+    const txHash = this.web3.utils.sha3(rawTx)
+
+    await sendRawTransaction(this.web3, rawTx)
+    await this.incrementTxCount(masterAddress)
+
+    logger.info(`Funded ${address} with ${txHash}`)
+
+    return txHash
+  }
+
+  /**
+   * Function triggered on every loop on the ongoing process loop we started in the constructor. Has
+   * lots of uses.
+   *
+   * - Prompt for funding of the master account
+   * - Announces a low balance
+   * - Funds children when necessary if autofundChildren was set
+   * - Does general tx tracking and pending tx count adjustments
+   * - Calls onReceipt callbacks if provided
+   */
+  async _process() {
+    let interval = 0
+    do {
+      if (!this.ready) continue
+
+      if (interval % 10 === 0) {
+        // Prompt for funding of the master account
+        const masterAddress = this.masterWallet.getChecksumAddressString()
+        const masterBalance = numberToBN(
+          await this.web3.eth.getBalance(masterAddress)
+        )
+        const balanceEther = this.web3.utils.fromWei(
+          masterBalance.toString(),
+          'ether'
+        )
+        if (masterBalance.eq(ZERO)) {
+          logger.error(
+            `Master account needs funding! Send funds to ${masterAddress}`
+          )
+        } else if (
+          masterBalance.lt(BASE_FUND_VALUE.mul(new BN(this.children.length)))
+        ) {
+          logger.warn(
+            `Master account is low @ ${balanceEther} Ether. Add funds soon!`
+          )
+        } else {
+          logger.info(`Master account balance: ${balanceEther} Ether`)
+        }
+
+        // Check for child balances dropping below set minimum and fund if necessary
+        if (this.ready && this.autofundChildren) {
+          for (let i = 0; i < this.children.length; i++) {
+            const child = this.children[i]
+            const childBalance = numberToBN(
+              await this.web3.eth.getBalance(child)
+            )
+            if (childBalance.lte(MIN_CHILD_BALANCE)) {
+              await this._fundChild(child)
+            }
+          }
+        }
+      }
+
+      /**
+       * Handle incoming receipts and remove pending transactions and adjust pending counts.
+       */
+      const pendingHashes = Object.keys(this.pendingTransactions)
+      for (const txHash of pendingHashes) {
+        const receipt = await this.web3.eth.getTransactionReceipt(txHash)
+
+        if (!receipt) continue
+
+        logger.debug(`Transaction ${txHash} has been mined.`)
+
+        // Remove from pending if it exists(it should)
+        if (!receipt.status) {
```

Comment:
Should this also check blockNumber is set ? I believe when I ran the June rewards payment we ran into this issue that Alchemy returns a receipt with no blockNumber before the tx is mined.